### PR TITLE
Added exception in destructor

### PR DESCRIPTION
**Pull request by andrewt0301**
Signed-off-by: andrewt0301 <andrewt0301@gmail.com>

### CODE REVIEW

File: false-negative/noexcept-crash.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    // If we throw exception directly, Clang-Tidy will report an error.
+    kill();
+}
+
+int main() {
+    try {
+        test();
+    } catch (...) {
+        // This line is never executed is exception is thrown in "noexcept"!
+        std::cout << "Exception is handled!";
```
Comment (yegor256) [inline false-negative/noexcept-crash.cpp]: @andrewt0301 let's get rid of IO in this class. it's not related to the error anyhow.

File: false-negative/noexcept-crash.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Clang-Tidy fails to detect an exception in a "nothrow" method
+// if the exception is thrown in a separate function.
+
+#include <iostream>
+
+// This method throws an exception.
+void kill() {
+    throw std::runtime_error("Fail!");
```
Comment (yegor256) [inline false-negative/noexcept-crash.cpp]: @andrewt0301 we can throw just an `int`, right? why do we need `std::` lib here?

**Comment by yegor256** [comment]
@andrewt0301 please, get rid of `iostream` -- it's irrelevant to the errors you describe

**Comment by andrewt0301** [comment]
done

**Comment by yegor256** [comment]
@andrewt0301 thanks!

**Pull request merged**